osc-udp.js

```diff
@@ -0,0 +1,105 @@
+const osc = require('osc');
+const dgram = require('dgram');
+const { onDataHandler } = require('./osc-feedback.js');
+
+class OSCUDPClient {
+	constructor(root, host, port, listen) {
+		this.root = root;
+		this.host = host;
+		this.port = port;
+		this.listen = listen;
+		this.udpPort = null;
+		this.connected = false;
+		this.socket = null;
+	}
+
+	openConnection() {
+		if (this.connected) {
+			this.root.log('info', 'UDP connection is already open');
+			return;
+		}
+
+		return new Promise((resolve, reject) => {
+			this.root.updateStatus('connecting');
+			this.socket = dgram.createSocket({ type: 'udp4', reuseAddr: true });
```

Comment:
is there a reason for doing this instead of using the udp socket type provided by the  `osc` library?

Reply:
The main reason is the SharedUdpSocket doesn't expose the full dgram library, opting for a [custom](https://bitfocus.github.io/companion-module-base/interfaces/SharedUdpSocket.html) implementation instead. I didn't really want to spend the time trying to reverse engineer it when I knew I had working code I could put in already. If this is a big issue, I can give it a further look however?